Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -1019,17 +1023,30 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
                         }
                         break;
                     case SceneQueryType.SphereOverlap:
-                        Collider[] colliders = UnityEngine.Physics.OverlapSphere(pointer.Rays[i].Origin, pointer.SphereCastRadius, ~UnityEngine.Physics.IgnoreRaycastLayer);
 
-                        if (colliders.Length > 0)
+                        int numColliders;
+
+                        if (colliderBuffer != null)
+                        {
+                            numColliders = UnityEngine.Physics.OverlapSphereNonAlloc(pointer.Rays[i].Origin, pointer.SphereCastRadius, colliderBuffer, ~UnityEngine.Physics.IgnoreRaycastLayer);
```

Comment:
I would suggest a simple buffer saturation check here like in #4942, so that users are notified in the unlikely case that the buffer is full and they may be missing interactions.

```
            if (numColliders == colliderBuffer.Length)
            {
                Debug.LogWarning($"Maximum number of {numColliders} colliders found in FocusProvider overlap query. Consider increasing the scene query buffer size in the pointer profile.");
            }
```